topics/imaging/tutorials/voronoi-segmentation/tutorial.md

```diff
@@ -0,0 +1,312 @@
+
+
+> <question-title></question-title>
+>
+> 1. What is the purpose of the smoothing step? 
+>
+> > <solution-title></solution-title>
+> >
+> > 1. The purpose of smoothing is to reduce noise. 
+> > For seed generation, noise might lead to false seeds where there is no object. 
+> > Smoothing also promotes connectedness within an object, where noise might make an object appear as two separate objects. 
+> > For mask generation, the same principles apply.
+> >
+> {: .solution}
+>
+{: .question}
+
+
+## Perform Voronoi segmentation based on the seeds 
+> <hands-on-title> Task description </hands-on-title>
+> We need to assign a unique label to each object in the seed image:
+> 1. {% tool [Convert binary image to label map](toolshed.g2.bx.psu.edu/repos/imgteam/binary2labelimage/ip_binary_to_labelimage/0.5+galaxy0) %} with the following parameters:
+>    - {% icon param-file %} *"Binary Image"*: `seeds`
+>    - *"Mode"*: `Connected component analysis`
+> 
+> 2. Rename the output to `label map`.
+>
+> 1. {% tool [Compute Voronoi tessellation with scikit-image](toolshed.g2.bx.psu.edu/repos/imgteam/voronoi_tesselation/voronoi_tessellation/0.22.0+galaxy3) %}. We use the label map to perform Voronoi segmentation. 
+>    - {% icon param-file %} *"Input Image"*: `label map`
+> 
+> 2. Rename the output to `tesselation`.
+>
+{: .hands_on}
+
+> <question-title></question-title>
+>
+> 2. How does the size of the seeds influence the Voronoi segmentation? 
+>
+> > <solution-title></solution-title>
+> >
+> > 2. A Voronoi segmentation partition a plane into regions based on [proximity to each member of a given set of objects](https://en.wikipedia.org/wiki/Voronoi_diagram). The algorithm is the same irregardless of whether the seeds are single points or objects with a spatial extension, but the size of the seeds will certainly alter the segmentation in some way. 
+> >
+> {: .solution}
+>
+{: .question}
+
+## Apply the mask and visualize the segmentation
+<!-- Combine the tesselation with the seeds and the mask to generate a better segmentation
+# Combine the tesselation with the mask to reduce the size of the Voronoi segments
+# Visualize the Voronoi segmentation by overlaying it on the original image -->
+A Voronoi tesselation segments an image into non-overlapping segments that cover the entire image. 
+This makes sure that segments do not overlap, but empty spaces between objects will be counted as part of the segment belonging to the nearest item. 
+A more accurate segmentation can be achieved by using the mask to reduce the size of the Voronoi segments. 
+This can be achieved with the following operation. 
+> <hands-on-title> Task description </hands-on-title>
+> Combine the tesselation with the seeds and the mask to generate a segmentation that limits the expanse of each segment: 
+> 1. {% tool [Process images using arithmetic expressions with numpy](toolshed.g2.bx.psu.edu/repos/imgteam/image_math/image_math/1.26.4+galaxy2) %} with the following parameters:
```

Comment:
The name of the tool is just `Process images using arithmetic expressions`, the `with NumPy` part is the description:
```suggestion
> 1. {% tool [Process images using arithmetic expressions](toolshed.g2.bx.psu.edu/repos/imgteam/image_math/image_math/1.26.4+galaxy2) %} with the following parameters:
```